src/content/docs/serverless-function-monitoring/aws-lambda-monitoring/get-started/compatibility-requirements-aws-lambda-monitoring.mdx

```diff
@@ -12,29 +12,18 @@ Before [enabling serverless monitoring](/docs/serverless-function-monitoring/aws
-* Python: `python3.7`, `python3.8`, `python3.9`, `python3.10`, `python3.11`
-* Go: `provided`, `provided.al2`
-* Java: `java8.al2`, `java11`
-* .NET Core: `dotnetcore3.1`
-
-AWS has older runtimes for these languages as well, but AWS has not [chosen to support](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html)
-the latest Lambda APIs with those older runtimes. Integration for older runtimes requires a different strategy, but
-is possible.
-
-Python and Node.js are by far the most popular languages in the Lambda ecosystem. The
-[New Relic Lambda Layers](https://layers.newrelic-external.com/)
-for Node.js and Python include the very latest New Relic agent version, and provide rich instrumentation with minimal
-configuration, right out of the box.
-
-Similarly, Go uses the New Relic Go agent. We recommend building self-hosting Go lambda functions, because the
-latest AWS Lambda APIs are not supported in the AWS `go1.x` runtime. Fortunately, this is well supported by the AWS
-Lambda SDK for Go. New Relic recommends keeping the agent module up to date. Support is limited for agent versions
-older than 3.9.0.
-
-To minimize performance impact, we've taken a different approach with Java and .NET Core. New Relic provides
-[OpenTracing](https://opentracing.io/) SDKs for these runtimes. This approach requires a bit more code to integrate.
-For newer runtimes, such as .NET 6, [Trace your .NET Lambda functions with New Relic and OpenTelemetry](/docs/serverless-function-monitoring/aws-lambda-monitoring/opentelemetry/lambda-opentelemetry-dotnet/) is available.
+* Node.js: `nodejs16.x`, `nodejs18.x`, `nodejs20.x`
+* Python: `python3.7`, `python3.8`, `python3.9`, `python3.10`, `python3.11`, `python3.12`
+* Go: `provided.al2`
+* Java: `java8.al2`, `java11`, `java17`
+
+AWS has older runtimes for these languages as well, but AWS has not [chosen to support](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) the latest Lambda APIs with those older runtimes. Integration for older runtimes requires a different strategy, but is possible.
+
+Python and Node.js are by far the most popular languages in the Lambda ecosystem. The [New Relic Lambda Layers](https://layers.newrelic-external.com/) for Node.js and Python include the very latest New Relic agent version, and provide rich instrumentation with minimal configuration, right out of the box.
+
+Similarly, Go uses the New Relic Go agent. New Relic recommends keeping the agent module up to date. Support is limited for agent versions older than 3.16.0.
+
+To minimize performance impact, we've taken a different approach with Java. New Relic provides the [OpenTracing](https://opentracing.io/) SDK for Java runtimes. This approach requires a bit more code to integrate. For newer runtimes, such as .NET 6, [Trace your .NET Lambda functions with New Relic and OpenTelemetry](/docs/serverless-function-monitoring/aws-lambda-monitoring/opentelemetry/lambda-opentelemetry-dotnet/) is available.
```

Comment:
@keegoid-nr this paragraph is confusing since the initial sentence seems to be talking about Java. Then, it switches abruptly to .NET with OpenTelemetry. So, is this paragraph about Java or about Java and .NET? 

Also, we have a separate document about using [Java with OpenTelemetry](https://deploy-preview-16445--docs-website-netlify.netlify.app/docs/serverless-function-monitoring/aws-lambda-monitoring/opentelemetry/lambda-opentelemetry-java/), which we don't seem to be linking to here. Could you clarify this?

Reply:
@rhetoric101 
The Java with OpenTelemetry doc is full of issues and it will be replaced with something better (coming soon).
`For newer runtimes` could be a new paragraph. We're trying to talk about all the runtimes on this page. However, it might be best to just delete that sentance entirely. .NET 6 is not longer a newer runtime and the lambda-opentelemetry-dotnet doc is also full of issues and will be updated with something better soon.

Reply:
@keegoid-nr I will delete this entire sentence: "For newer runtimes, such as .NET 6, [Trace your .NET Lambda functions with New Relic and OpenTelemetry](/docs/serverless-function-monitoring/aws-lambda-monitoring/opentelemetry/lambda-opentelemetry-dotnet/) is available."